sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -159,13 +163,104 @@ pub fn test_exit(
     Ok(())
 }
 
+// Execute an openssl invocation, passing the args[] as command line parameters.
+// The intended use is openssl x509 certificate verification. cert_num is the
+// number of the certificate in the list of certificates being validated.
+fn openssl_command(cert_num: usize, args: &[&str]) -> Result<()> {
+    let o = Command::new("openssl").args(args).output()?;
+    if !o.status.success() {
+        log::error!(
+            "openssl output:\n{}",
+            std::str::from_utf8(&o.stderr).unwrap()
+        );
+        Err(anyhow!("Cert #{cert_num}: openssl {} failed", args[0]))
+    } else {
+        Ok(())
+    }
+}
+
+// Given a u8 blob containing an x509 certificate perform some rudimentary
+// header correctness checks and return the actual certificate size based on the
+// ASN.1 header length field contents.
+fn get_cert_size(cert: &[u8]) -> Result<usize> {
+    let len = cert.len();
+
+    if len < 4 {
+        return Err(anyhow!("Certificate too short {len}"));
+    }
+
+    if cert[0] != 0x30 || cert[1] != 0x82 {
+        Err(anyhow!(
```

Comment:
You can keep it like this but also anyhow provide a macro that does exactly this: `bail!`.